tests/TestCase/ORM/Association/HasManyTest.php

```diff
@@ -643,4 +643,75 @@ protected function assertSelectClause($expected, $query)
         }
         $this->assertEquals($expected, $query->clause('select'));
     }
+
+    /**
+     * Tests that unlinking calls the right methods
+     *
+     * @return void
+     */
+    public function testUnlinkSuccess()
+    {
+        $assoc = $this->author->hasMany('Articles', [
+            'sourceTable' => $this->author,
+            'targetTable' => $this->article
+        ]);
+
+        $entity = $this->author->get(1, ['contain' => 'Articles']);
+        $initial = $entity->articles;
+        $this->assertCount(2, $initial);
+
+        $this->author->unlink($entity, $entity->tags);
+        $this->assertEmpty($entity->get('articles'), 'Property should be empty');
+
+        $new = $this->article->get(2, ['contain' => 'Articles']);
+        $this->assertCount(0, $new->articles, 'DB should be clean');
+        $this->assertSame(4, $this->author->find()->count(), 'Authors should still exist');
+        $this->assertSame(3, $this->article->find()->count(), 'Articles should still exist');
+    }
+
+    /**
+     * Tests that unlink will work without contain() being called
+     */
+    public function testUnlinkWithoutContain()
```

Comment:
I'll be removing this test as it is invalid. The method signature for unlink() requires that `$targetEntities` be an array, which a Query object is not.